django_filters/compat.py

```diff
@@ -1,5 +1,16 @@
 from django.conf import settings
 
+try:
+    from django.db.models.sql.constants import QUERY_TERMS
+except ImportError:
+    # Django 2.1+ does not have QUERY_TERMS anymore
+    QUERY_TERMS = {
+        'contains', 'day', 'endswith', 'exact', 'gt', 'gte', 'hour',
+        'icontains', 'iendswith', 'iexact', 'in', 'iregex', 'isnull',
+        'istartswith', 'lt', 'lte', 'minute', 'month', 'range', 'regex',
+        'search', 'second', 'startswith', 'week_day', 'year',
+    }
```

Comment:
I guess this could even live in `constants.py` without trying to import the constant from django? What do you think?

Reply:
`QUERY_TERMS`/`LOOKUP_TYPES` are only used for `LookupTypeField`. They're not really used elsewhere in the library or intended to be used by end users, so `compat` makes sense.

Also, #851 completely reworks the 'multiple lookup types' handling, and indirectly fixes 2.1 compatibility, as it removes reliance on `QUERY_TERMS`.

Reply:
Wasn't aware of #851. Is it getting merged anytime soon? If so, I guess it's fine to close this one without merging?

Reply:
I'd leave this open for now. #851 doesn't fix the tests, and there's no guarantee that it will get merged. Also, it won't be too difficult to rebase that PR off of this if necessary.

It might be a while before these PRs are merged. Carlton's mostly been focusing on DRF and Django as of late. Maintenance has mostly halted since the 2.0.0.dev1 release back in October.

I'm probably going to pester @carltongibson soon 😉

Reply:
Secret: I'm waiting on DRF v3.9 before taking on the backlog here. That's deliberate. Django Filter 2.0 will tie-in with the changes in DRF. Django Filter is super-mature, so I've not seen any urgency. 

Don't close anything as yet. I'd rather review it all in one batch. (I'm picking off the obvious ones as we go.) 

Thanks for the input all!